code/components/discord/src/DiscordPresence.cpp

```diff
@@ -162,6 +170,7 @@ static InitFunction initFunction([]()
 	g_discordAppAssetSmall = DEFAULT_APP_ASSET_SMALL;
 	g_discordAppAssetText = DEFAULT_APP_ASSET_TEXT;
 	g_discordAppAssetSmallText = DEFAULT_APP_ASSET_SMALL_TEXT;
+	g_richPresenceEnabledVar = std::make_shared<ConVar<bool>>("game_enableDiscordRichPresence", ConVar_Archive | ConVar_UserPref, g_richPresenceEnabled, &g_richPresenceEnabled);
```

Comment:
There's an overload on `ConVar` to add a callback for changes, there you can do (un)initialization and (un)register the update function (see `fwEvent<T>::Connect` and `fwEvent<T>::Disconnect(cookie)` on `OnGameFrame`).

Will remove the per frame overhead.

Reply:
This would still have overhead per-frame anyways because you would have to do relatively the same check  inside of `UpdatePresence` to make sure it doesn't try to update rich presence.

Reply:
I don't see how that adds up. You check for `g_richPresenceEnabled` and return if it's not true, which would not call `UpdatePresence()` at all. Unregistering the this update call will result in the same behavior, just with less calls and checks

Reply:
You're right, I completely glossed over you mentioning disconnecting the event.

Reply:
This may be resolved with the most recent push, hopefully, implemented like expected.

Reply:
Looking nice, I'll need to confirm that this is being called at startup as well, I'll probably get to that next week or maybe someone else will get to that sooner.

Another thing I'm considering is if `game_` should be changed into something like `cl_` as `game_` is used for game (GTAV/RDR2) related features.

Reply:
Tested startup behavior, should work. Let me know if anything else should be changed :) Thanks you!